CONTRIBUTING.md

```diff
@@ -74,38 +64,66 @@ The following guidelines should be observed when creating a PR.
+* Try to keep contributions small, as this speeds up the reviewing process. In the case of large contributions, e.g. a new complex `EOTask`, it's best to contact us first to review the scope of the contribution.
 
- * Keep API compatibility in mind, in particular when contributing a new `EOTask`. In general,
-   all new tasks should adhere to the modularity of **eo-learn**.
-   Check the Section below for more information on how to contribute an `EOTask`.
+* Keep API compatibility in mind, in particular when contributing a new `EOTask`. In general, all new tasks should adhere to the modularity of **eo-learn**. Check the Section below for more information on how to contribute an `EOTask`.
 
- * New features or tasks should be appropriately commented using Sphinx style docstrings. The documentation uses
-   the [PEP-8][pep-8] formatting guidelines. [Pylint][pylint] is used to check the coding standard.
-   Therefore please run `pylint` from the the main folder, which contains the `pylintrc` file, to make sure your
-   contribution is scored `10.0/10.0`.
+* New features or tasks should be appropriately commented using Sphinx style docstrings. The documentation uses the [PEP-8][pep-8] formatting guidelines. [Pylint][pylint] is used to check the coding standard. Therefore, please run `pylint */ *.py` from the the main folder, which contains the `pylintrc` file, to make sure your contribution is scored `10.0/10.0`.
+
+### Development environment
+
+* Get the latest development version by creating a fork and clone the repo:
+
+```bash
+git clone git@github.com:<username>/eo-learn.git
+```
+
+* Make sure that you have python 3 installed, i.e. version 3.5 or higher. See https://www.python.org/downloads/ for general Python versions or https://www.anaconda.com/distribution/ for using Conda.
+
+* Install **eo-learn** and all subpackages in (editable) development mode with `pip install -e <package_folder>`. We strongly recommend initializing a virtual environment before installing the required packages. For example by using Python Package Index (PyPI) and virtualenv:
+
+```bash
+cd eo-learn
+# The following creates the virtual environment in the ".env" folder.
+virtualenv --python python3 .env
+source .env/bin/activate
+
+# The following installs all eo-learn subpackages and development packages
+# using PyPI in the activated virtualenv environment.
+pip install -e ./core -e ./coregistration -e ./features -e ./geometry -e ./io \
+-e ./mask -e ./ml_tools -e ./visualization -e .
+pip install -r requirements-dev.txt -r requirements-docs.txt
+```
+
+Or alternatively by using Conda:
+
+```bash
+cd eo-learn
+# The following creates the virtual environment named "dev_eolearn" where the
+# Conda installation in located. ipykernel enables Jupyter (using
+# nb_conda_kernels) to select the environment as a kernel. ipywidgets enables
+# Jupyter to show the progress bar of EOExecutor without raising an error.
+conda create -n dev_eolearn python=3.7 ipykernel ipywidgets
```

Comment:
@mlubej. I'm not sure I understand what you wish. Is it these installed Conda packages: `ipykernel` and `ipywidgets`?